### PR TITLE
issues with 404 without index.php

### PR DESCRIPTION
**Pull request by danrcoull**
### Description
unset index, on route check return false if not required. 
check the request uri to make sure nothing is passed or if admin is passed

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=18378) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=18378) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: danrcoull<br/>:x: dan coull<hr/>**dan coull** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

File: phpserver/router.php [hunk 1]
```diff
@@ -39,13 +39,23 @@
 
 if (php_sapi_name() === 'cli-server') {
     $debug("URI: {$_SERVER["REQUEST_URI"]}");
+    if($_SERVER["REQUEST_URI"] == '/'|| strpos($_SERVER["REQUEST_URI"],'/admin') !==false){
```
Comment (sidolov) [inline phpserver/router.php]: We can't hardcode `/admin` path because it can be different according to store settings

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by sidolov** [comment]
Hi @danrcoull  , please, sign CLA, otherwise, we can't process your pull request

**Comment by danrcoull** [comment]
signed.

File: phpserver/router.php (hunk 1 above)
Comment (danrcoull) [reply phpserver/router.php]: Though we could pull directly from the config like below. 
```
$admin = isset($etcConfig['backend']['frontName']) ? $etcConfig['backend']['frontName'] : 'NO-CONFIG';

 if($_SERVER["REQUEST_URI"] == '/'|| strpos($_SERVER["REQUEST_URI"],$admin) !==false)
 {
    return false;
}
```

**Comment by danrcoull** [comment]
Ive made some modds to the code, to check the config and check if it exists to get the backend name.

**Pull request closed without merging**